File: PSA/SmartAisle/app/WelcomeScreen.js
import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

const WelcomeScreen = ({ navigate }) => {
    return (
        <LinearGradient
            colors={['#87CEEB', '#87CEFA']}
            style={styles.container}
        >
            <Text style={styles.title}>Welcome</Text>
            <Text style={styles.subtitle}>Do you have a SmartAisle account?</Text>
            <View style={styles.buttonContainer}>
                <TouchableOpacity
                    style={styles.button}
                    onPress={() => navigate('LoginScreen')}
                    accessibilityLabel="Navigate to Login Screen"
                >
                    <Text style={styles.buttonText}>Yes</Text>
                </TouchableOpacity>
                <TouchableOpacity
                    style={styles.button}
                    onPress={() => navigate('RegistrationScreen')}
                    accessibilityLabel="Navigate to Registration Screen"
                >
                    <Text style={styles.buttonText}>No</Text>
                </TouchableOpacity>
            </View>
        </LinearGradient>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    title: {
        fontSize: 30,
        fontWeight: 'bold',
        color: 'white',
        marginBottom: 20,
    },
    subtitle: {
        fontSize: 18,
        color: 'white',
        marginBottom: 40,
        textAlign: 'center',
    },
    buttonContainer: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        width: '80%',
    },
    button: {
        backgroundColor: 'white',
        padding: 15,
        width: 100,
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center',
        marginHorizontal: 10,
    },
    buttonText: {
        fontSize: 16,
        fontWeight: 'bold',
        color: '#87CEEB',
    },
});

export default WelcomeScreen;

File: SmartAisle/WelcomeScreen.swift
import SwiftUI

struct WelcomeScreen: View {
    @Binding var navigate: Screen
    
    var body: some View {
        LinearGradient(gradient: Gradient(colors: [Color(red: 0.53, green: 0.81, blue: 0.92), Color(red: 0.53, green: 0.82, blue: 0.98)]), startPoint: .top, endPoint: .bottom)
            .edgesIgnoringSafeArea(.all)
            .overlay(
                VStack {
                    Text("Welcome")
                        .font(.largeTitle)
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                        .padding(.bottom, 20)
                    
                    Text("Do you have a SmartAisle account?")
                        .font(.title2)
                        .foregroundColor(.white)
                        .padding(.bottom, 40)
                        .multilineTextAlignment(.center)
                    
                    HStack(spacing: 20) {
                        Button(action: {
                            navigate = .login
                        }) {
                            Text("Yes")
                                .fontWeight(.bold)
                                .foregroundColor(Color(red: 0.53, green: 0.81, blue: 0.92))
                                .padding()
                                .frame(width: 100)
                                .background(Color.white)
                                .cornerRadius(10)
                        }
                        
                        Button(action: {
                            navigate = .registration
                        }) {
                            Text("No")
                                .fontWeight(.bold)
                                .foregroundColor(Color(red: 0.53, green: 0.81, blue: 0.92))
                                .padding()
                                .frame(width: 100)
                                .background(Color.white)
                                .cornerRadius(10)
                        }
                    }
                }
            )
    }
}

struct WelcomeScreen_Previews: PreviewProvider {
    static var previews: some View {
        WelcomeScreen(navigate: .constant(.welcome))
    }
}
